### DISCUSSION TITLE
Enabling autocomplete for a custom language using Rust grammar

### DISCUSSION THREAD

**Discussion by selimsandal** [helix-editor/helix#11907] [Troubleshooting] [answered]
I am trying to add support for [DSLX](https://github.com/google/xls), the language uses mostly Rust syntax so I used Rust grammar for highlighting and added dslx_ls as the language server. Language server features work but autocomplete does not. Is there a way I can make autocomplete work using `dslx_ls` or reuse Rust autocomplete?

Here is my config:

**languages.toml**
```
[[language]]
name = "dslx"
scope = "source.dslx"
file-types = ["x"]
injection-regex = "dslx"
language-servers = [ "dslx_ls" ]
grammar = "rust"
indent = { tab-width = 2, unit = "  " }

[language-server.dslx_ls]
command = "dslx_ls"
```

**runtime/queries/dslx/highlights.scm**
```
; inherits: rust
```

**Comment by seifertm** [comment, marked as answer]
It looks like _dslx_ls_ does not support autocompletion.

Looking at the linked repository I found [the language server implementation](https://github.com/google/xls/blob/0484d66165d67de49ee35212ba3920c1b782f8fe/xls/dslx/lsp/dslx_ls.cc). It doesn't seem to implement a handler for [textDocument/completion](https://microsoft.github.io/language-server-protocol/specifications/lsp/3.17/specification/#textDocument_completion), which is used to retrieve a list of possible completions from a language server.
